yt/frontends/arepo/io.py

```diff
@@ -5,6 +5,10 @@
 class IOHandlerArepoHDF5(IOHandlerGadgetHDF5):
     _dataset_type = "arepo_hdf5"
 
+    def _generate_smoothing_length(self, data_files, kdtree):
```

Comment:
if this empty function necessary ? I can't find where it's called

Reply:
This is called from `SPHParticleIndex._initialize_index`. If we can't find smoothing lengths, we need to generate them. However, in the Arepo frontend we want to generate them a different way than in the case being handled in this PR. However, there is probably a better way to do this so I will see what I can do.

Reply:
Just so I am sure I understand -- does this mean that when setting up the index, we don't have the smoothing lengths?  So we won't set the ghost zones needed in the index precisely right?

Reply:
@matthewturk I thought about this and it doesn't mean that. it just means that in the Arepo case we don't have to generate the lengths since we're just computing them from the volume.